web/src/components/pages/dashboard/challenge/weekly-challenge.tsx

```diff
@@ -13,132 +18,195 @@ const WEEKS: Array<string> = [
-      pastWeek.push(currentWeek - 2);
-      break;
-  }
-  const label = pastWeek.length === 0 ? 'Future' : 'Past';
+interface Props extends PropsFromState {
+  isNarrow?: boolean;
+}
 
-  const api = useAPI();
-  const account = useAccount();
+interface State {
+  weekly: WeeklyChallenge;
+}
 
-  const [weekly, setWeekly] = useState({
-    week: 1,
-    user: {
-      speak: 50,
-      speak_total: 200,
-      listen: 25,
-      listen_total: 100,
-    },
-    team: {
-      invite: 50,
-      invite_total: 200,
-    },
-  });
-  useEffect(() => {
-    api.fetchWeeklyChallenge().then(setWeekly);
-  }, []);
-  return (
-    <div className="weekly-container">
-      <div className="weekly-topbar">
-        <h2>Weekly Challenge</h2>
-        <div className="weeks">
-          <span className="week">Week</span>
-          {WEEKS.map((title, index) => (
-            <span
-              key={index}
-              className={`week-number ${
-                currentWeek === index ? 'active' : ''
-              }`}>
-              {++index}
-            </span>
-          ))}
-        </div>
-        <div className="week-points">
-          <p>
-            <img src={require('./images/star.svg')} alt="score" />
-            <span>+50 points</span>
-          </p>
-          <p>Way to send your first invite!</p>
-        </div>
-      </div>
-      <div className="weekly-content">
-        <div>
-          <p className="weekly-title">Current challenge</p>
-          {weekly && (
-            <WeeklyChallengeBoard
-              title={WEEKS[currentWeek]}
-              week={currentWeek}
-              isNarrow={isNarrow}
-              avatarUrl={account.avatar_url}
-              logoUrl={challengeLogoUrls[account.enrollment.team]}
-              weekly={weekly}
-            />
-          )}
-        </div>
-        {pastWeek.length !== WEEKS.length - 1 && (
-          <div>
-            <p className="weekly-title">Next challenge</p>
-            {weekly && (
-              <WeeklyChallengeBoard
-                isDisabled
-                title={WEEKS[currentWeek + 1]}
-                week={currentWeek + 1}
-                isNarrow={isNarrow}
-                avatarUrl={account.avatar_url}
-                logoUrl={challengeLogoUrls[account.enrollment.team]}
-                weekly={weekly}
-              />
-            )}
+class WeeklyChallengeCard extends React.Component<Props, State> {
+  state: State = {
+    weekly: null,
+  };
+  getCurrentWeek = (): void => {
+    let now = moment().week();
+    let startWeek = moment(START_DATE).week();
+    this.currentWeek = now - startWeek;
+    switch (this.currentWeek) {
+      case 0:
+        break;
+      case 1:
+        this.pastWeek.push(this.currentWeek - 1);
+        break;
+      case 2:
+        this.pastWeek.push(this.currentWeek - 1);
+        this.pastWeek.push(this.currentWeek - 2);
+        break;
+    }
+  };
+  currentWeek = 0;
+  pastWeek: Array<number> = [];
```

Comment:
I think we should reuse the recent `weekly.week` data for this, so we're not doing any timekeeping on the client. We definitely don't want to use `moment` here!